File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/CampusRepositoryImpl.kt
package com.ssafy.ssabree.core.repository

import com.ssafy.ssabree.core.datasource.remote.CampusService
import com.ssafy.ssabree.core.datasource.remote.model.Ban
import com.ssafy.ssabree.core.datasource.remote.model.Campus
import com.ssafy.ssabree.core.utils.RetrofitClient

class CampusRepositoryImpl : CampusRepository {

    private val campusService = RetrofitClient.instance.create(CampusService::class.java)

    override suspend fun getCampuses(): Result<List<Campus>> {
        return runCatching {
            campusService.getCampuses()
        }
    }

    override suspend fun getClasses(campusId: Int): Result<List<Ban>> {
        return runCatching {
            campusService.getClasses(campusId)
        }
    }
}

File: iOS/ssabree/ssabree/Core/Repository/CampusRepositoryImpl.swift
import Foundation

final class CampusRepositoryImpl: CampusRepository {
    private let campusService: CampusService

    init(campusService: CampusService) {
        self.campusService = campusService
    }

    func getCampuses() async -> Result<[Campus], Error> {
        do {
            let response = try await campusService.getCampuses()
            return .success(response)
        } catch {
            return .failure(error)
        }
    }

    func getClasses(campusId: Int) async -> Result<[Ban], Error> {
        do {
            let response = try await campusService.getClasses(campusId: campusId)
            return .success(response)
        } catch {
            return .failure(error)
        }
    }
}
